File: tomatomall-frontend/src/api/review.ts
import { axios } from "@/utils/request";

export interface Review {
    id: number;
    accountId: number;
    content: string;
    rate: number;
    type: reviewType;
    productId: number;
    shopId: number;
    createdAt: string;
}

export type reviewType = 'PRODUCT' | 'SHOP';


export const addProductReview = async (review: Review) => {
    return await axios.post(`/api/reviews/product`, review, {
        headers: { "Content-Type": "application/json" },
    });
}

export const addShopReview = async (review: Review) => {
    return await axios.post(`/api/reviews/shop`, review, {
        headers: { "Content-Type": "application/json" },
    });
}

export const getProductReviews = async (productId: number) => {
    return await axios.get(`/api/reviews/product/${productId}`);
}

export const getShopReviews = async (shopId: number) => {
    return await axios.get(`/api/reviews/shop/${shopId}`);
}

export const deleteReview = async (reviewId: number) => {
    return await axios.delete(`/api/reviews/${reviewId}`);
}

File: tomatomall-backend/src/main/java/com/example/tomatomall/po/Review.java
package com.example.tomatomall.po;

import com.example.tomatomall.vo.ProductVO;
import com.example.tomatomall.vo.ReviewVO;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.beans.BeanUtils;

import javax.persistence.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Getter
@Setter
@NoArgsConstructor
@Entity
@Table(name = "reviews")
public class Review {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "account_id", nullable = false)
    private Integer accountId;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String content;

    @Column(nullable = false, precision = 3, scale = 2)
    private BigDecimal rate;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ReviewType type;

    @Column(name = "product_id")
    private Integer productId;

    @Column(name = "shop_id")
    private Integer shopId;

    @Column(name = "created_at", insertable = false, updatable = false)
    private LocalDateTime createdAt;

    public enum ReviewType {
        PRODUCT,
        SHOP
    }

    public ReviewVO toVO() {
        ReviewVO vo = new ReviewVO();
        vo.setId(id);
        vo.setAccountId(accountId);
        vo.setContent(content);
        vo.setRate(rate);
        vo.setType(type.toString());
        vo.setProductId(productId);
        vo.setShopId(shopId);
        vo.setCreatedAt(createdAt);
        return vo;
    }
}